Add support for LoongArch.

liuxiang88 opened the pull request:
This commit is adding support for LoongArch architecture.

selmf commented:
Thank you for the PR. I see no issue in merging this, but it would be great if you could ask the upstream project for the LZMA sdk (7zip) if they are interested in adding LoongArch support.

You can find the project here: https://sourceforge.net/p/sevenzip/patches/

I also need some general information like your name and an email to add you to the list of authors. Of course, if you'd prefer to contribute anonymous we can do that too.

liuxiang88 commented:
Thank you!
My name is liuxiang,Email is liuxiang@loongson.cn.

selmf commented:
Ok, I have added you to AUTHORS and updated the changelog. Could you take a quick look and tell me if everything is correct?

liuxiang88 commented:
> Ok, I have added you to AUTHORS and updated the changelog. Could you take a quick look and tell me if everything is correct?

Thank you.I've already seen it and there's no problem.Is it time to merge my commit?

selmf commented:
Merged, thank you for the contribution.

liuxiang88 commented on lzmasdk/CpuArch.h:
```diff
@@ -83,6 +83,14 @@ MY_CPU_LE_UNALIGN means that CPU is LITTLE ENDIAN and CPU supports unaligned mem
 
 
+#if  defined(__loongarch64) \
+  || defined(__loongarch__) \
```
Yes,you are right.After the GCC version is updated, the definition changes.I'll make another commit to update.

zjiaz replied:
Okay. And I thought I have left the comment before the merge, but actually it was just pending in my timeline. So sorry for the late review.
